### DISCUSSION TITLE
chanserv protection

### DISCUSSION THREAD

**Discussion by medkinson** [anope/anope#460] [Support] [answered]
hi! how to protect channel from unauthorized users (nickserv IDENTIFY and recognized)?
that should doesn't pass them into chan, saying something like: Cannot join channel (Need to be identified and verified to join this channel, '/msg NickServ help' to learn how to register and verify.) (like in tor browser's IRC - oftc #tor)

**Comment by medkinson** [comment]
and i am using qwebirc as a client and inspircd as a server

**Reply by medkinson** [reply]
and DO not ban them

**Comment by sadiepowell** [comment]
You want the reginvite (+R) or regmoderated (+M) channel modes.

https://docs.inspircd.org/4/modules/account/#channel-modes

https://docs.inspircd.org/3/modules/services_account/#channel-modes

**Reply by medkinson** [reply]
thanks, and how i can make the chanserv automatically join that user to chosen channel

**Reply by sadiepowell** [reply, marked as answer]
You can't do that with services. You can do it at the IRCd level though.

https://docs.inspircd.org/3/modules/conn_join/

https://docs.inspircd.org/4/modules/conn_join/
